### ISSUE TITLE
clink + Win10 + cyrillic characters with win console V2 - doesn't work 

### ISSUE THREAD

**Issue by Slach** [chrisant996/clink#1]
thanks a lot for your efforts with https://github.com/chrisant996/clink/releases/tag/v1.1.0.4b5991 ;)
i just create issue on this fork
for notify
https://github.com/mridgers/clink/issues/544, still reproduce on your last release ;)

**Comment by chrisant996** [comment]
Thanks!  Yes, I haven't worked on that yet.  My intent is to include a fix in the first Beta build, provided a fix doesn't prove too invasive (in which case it might get deferred a little bit).

**Comment by chrisant996** [comment]
I'm working on it now, but I'm running into a number of issues with the GNU Readline library along the way.  This could take a while to get working properly, but it will get there eventually.

**Comment by chrisant996** [comment]
I observed several different strange problems with Unicode input.  It took me a combined total of around 7 hours of debugging through low level details of Readline and Clink UTF8 processing for the various problems to finally track down the root cause:

Clink forgot to tell the C runtime that it wanted to use UTF8.

Readline asks the C runtime what locale to use, and the C runtime defaults to using the "minimal ANSI conforming environment for C translation", which is essentially an extended-ASCII pseudo-locale.  So Clink was trying to talk UTF8, and Readline was trying to talk pseudo-ASCII.  No wonder Unicode input went awry.

A one line change seems to fix everything (I'm testing further to confirm):
```cpp
        // Readline needs to know it should use UTF8.
        setlocale(LC_ALL, ".utf8");
```

**Comment by chrisant996** [comment]
Fixed by [004c691](004c691aac6f089c375b773dee9ed8e98baa4017).
Binaries available [here](https://github.com/chrisant996/clink/actions/runs/300038802).

**Comment by Slach** [comment]
@chrisant996 thanks a lot for your efforts, all works fine now
